### DISCUSSION TITLE
Localization in TabbedPage children

### DISCUSSION THREAD

**Discussion by Pipozor** [SirJohnK/LocalizationResourceManager.Maui#27] [Q&A] [answered]
Hi,
First, congratulations for this great extension. Great work.

My MainPage is a TabbedPage.
I do the culture change in a children AccountPage of TabbedPage. The culture change works well on AccountPage  tab. But not on all other children tabs. I don't know if there is a way "to refresh, to notify or to invoke" other tabs to refresh their culture?

```
<TabbedPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
            xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
            xmlns:localization="clr-namespace:LocalizationResourceManager.Maui;assembly=LocalizationResourceManager.Maui"
            x:Class="Test.MainPage">
    
    <TabbedPage.Children>
        <local:HomePage IconImageSource="page_icon_home.png" />
        <local:HistoryPage IconImageSource="page_icon_history.png" />
        <local:AccountPage IconImageSource="page_icon_account.png" />
    </TabbedPage.Children>

</TabbedPage>
```

On AccountPage.xaml.cs it is working (texts changed instantly):

```
public partial class AccountPage : ContentPage
{
       public AccountPage()
       {
           InitializeComponent();
           BindingContext = this;
       }
        private async void OnUpdateClicked(object sender, EventArgs e)
        {
           .../....
           MauiProgram.ResourceManager.CurrentCulture = new CultureInfo(UserAccount.DefaultLanguage);
        }
}
```

Please remark I cannot inject ILocalizationResourceManager in MainPage(...) or a ViewModel(...) or any children page for various reasons... 
I am using a global injection from MauiProgram.cs to use ILocalizationResourceManager.

```
    public static class MauiProgram
    {

        public static ILocalizationResourceManager ResourceManager;
       

        public static MauiApp CreateMauiApp()
        {

            var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                .UseMauiCommunityToolkit()
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                })
                .UseLocalizationResourceManager(settings =>
                {
                    settings.AddResource(AppResources.ResourceManager);
                    settings.RestoreLatestCulture(true);
                });

                     var host = builder.Build();

            ResourceManager = host.Services.GetService<ILocalizationResourceManager>();

            return host;
        }
    }
```

**Comment by SirJohnK** [comment]
@Pipozor , will look into this asap! 😃

**Reply by Pipozor** [reply]
Thank you Sir.
In the meantime I used a working (dirty) trick on each tab, here a sample :  
```
namespace LiDi
{
    using Microsoft.Maui.Controls;
    using System;

    public partial class HistoryPage : ContentPage
    {
        protected override async void OnAppearing()
        {
            base.OnAppearing();
            Localization.CheckCultureChange();
        }
    }
}
```

And in my Localization class, a simple method :

        public static void CheckCultureChange()
        {
            if (string.IsNullOrEmpty(App.BaseViewModel.UserAccount.DefaultLanguage))
            {
                return;
            }
            var culture = new CultureInfo(App.BaseViewModel.UserAccount.DefaultLanguage);
             MauiProgram.ResourceManager.CurrentCulture = culture;
        }

**Comment by SirJohnK** [comment, marked as answer]
@Pipozor , I finally had the time to look into this. 😄
I have tried to recreate your issue in an new Maui project with TabbedPage and static ResourceManager, but for me it works just fine!
Look at this repo: [https://github.com/SirJohnK/LocalizeTabbedPagesTest](https://github.com/SirJohnK/LocalizeTabbedPagesTest)

**Reply by SirJohnK** [reply]
@Pipozor , did you have time to look at this?😃

**Comment by Pipozor** [comment]
Hi Johan,

Sorry I was on holiday. I just saw your email.
With the trick I did it works. But not by default. But you tried and said
it is working fine on your end.
I will try to investigate more and let you know. I probably made a mistake.

Cheers,
David

Le mar. 20 août 2024 à 21:44, Johan Svensson ***@***.***> a
écrit :

**Reply by SirJohnK** [reply]
Ok. Download the test repo that I did and verify that you have done the same. 
[https://github.com/SirJohnK/LocalizeTabbedPagesTest](https://github.com/SirJohnK/LocalizeTabbedPagesTest)

**Comment by Pipozor** [comment]
Sorry for the late reply.
Finally, the issue did not appeared anymore. Probably an issue from my code.

**Comment by SirJohnK** [comment]
Ok, great! I will close this discussion. 👍
